html2docx/image.py

```diff
@@ -0,0 +1,82 @@
+import io
+import pathlib
+import time
+import urllib.error
+import urllib.request
+from typing import Dict, Optional
+
+from docx.image.exceptions import UnrecognizedImageError
+from docx.image.image import Image
+from docx.shared import Inches
+
+USABLE_HEIGHT = Inches(8.1)
+USABLE_WIDTH = Inches(5.8)
+
+MAX_IMAGE_SIZE = 10 * 1024 * 1024  # 10 MiB
+
+
+def load_image(src: str) -> io.BytesIO:
+    image_buffer = None
+    retry = 3
+    while retry and not image_buffer:
+        try:
+            with urllib.request.urlopen(src) as response:
+                size = response.getheader("Content-Length")
+                if size and int(size) > MAX_IMAGE_SIZE:
+                    break
+                data = response.read(MAX_IMAGE_SIZE + 1)
+        except (ValueError, urllib.error.HTTPError):
+            retry = 0
+        except urllib.error.URLError:
+            retry -= 1
+            if retry:
+                time.sleep(1)
+        else:
+            if len(data) <= MAX_IMAGE_SIZE:
+                image_buffer = io.BytesIO(data)
+
+    if image_buffer:
+        try:
+            Image.from_blob(image_buffer.getbuffer())
+        except UnrecognizedImageError:
+            image_buffer = None
+
+    if not image_buffer:
+        broken_img_path = pathlib.Path(__file__).parent / "image-broken.png"
+        image_buffer = io.BytesIO(broken_img_path.read_bytes())
+
+    return image_buffer
+
+
+def image_size(
```

Comment:
Could you add a doc-string to this function or at least a doc test.

Reply:
Good idea, I wrote a docstring. Please let me know if you’d like to see anything else there.